File: Snooper/src/test/java/com/prateekj/snooper/networksnooper/viewmodel/HttpHeaderViewModelTest.java
package com.prateekj.snooper.networksnooper.viewmodel;

import com.prateekj.snooper.networksnooper.model.HttpHeader;
import com.prateekj.snooper.networksnooper.model.HttpHeaderValue;

import org.junit.Before;
import org.junit.Test;

import static java.util.Arrays.asList;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class HttpHeaderViewModelTest {

  private HttpHeaderViewModel httpHeaderViewModel;

  @Before
  public void setUp() throws Exception {
    HttpHeader httpHeader = new HttpHeader("accept-language");
    HttpHeaderValue value1 = new HttpHeaderValue("en-US,en");
    HttpHeaderValue value2 = new HttpHeaderValue("q=0.8,hi");
    HttpHeaderValue value3 = new HttpHeaderValue("q=0.6");
    httpHeader.setValues(asList(value1, value2, value3));
    httpHeaderViewModel = new HttpHeaderViewModel(httpHeader);
  }

  @Test
  public void shouldReturnHeaderName() throws Exception {
    assertThat(httpHeaderViewModel.headerName(), is("accept-language"));
  }

  @Test
  public void shouldReturnHeaderValues() throws Exception {
    assertThat(httpHeaderViewModel.headerValues(), is("en-US,en;q=0.8,hi;q=0.6"));
  }
}

File: Snooper/src/test/java/com/prateekj/snooper/networksnooper/viewmodel/HttpHeaderViewModelTest.kt
package com.prateekj.snooper.networksnooper.viewmodel

import com.prateekj.snooper.networksnooper.model.HttpHeader
import com.prateekj.snooper.networksnooper.model.HttpHeaderValue

import org.junit.Before
import org.junit.Test

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat

class HttpHeaderViewModelTest {

  private lateinit var httpHeaderViewModel: HttpHeaderViewModel

  @Before
  @Throws(Exception::class)
  fun setUp() {
    val httpHeader = HttpHeader("accept-language")
    val value1 = HttpHeaderValue("en-US,en")
    val value2 = HttpHeaderValue("q=0.8,hi")
    val value3 = HttpHeaderValue("q=0.6")
    httpHeader.values = listOf(value1, value2, value3)
    httpHeaderViewModel = HttpHeaderViewModel(httpHeader)
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnHeaderName() {
    assertThat(httpHeaderViewModel.headerName(), `is`("accept-language"))
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnHeaderValues() {
    assertThat(httpHeaderViewModel.headerValues(), `is`("en-US,en;q=0.8,hi;q=0.6"))
  }
}